### PR TITLE
Auth Layout - ClientApp

### PR DESCRIPTION
**Pull request by KryaKa57**

### CODE REVIEW

**Comment by iPakTulane** [comment]
Overall, it is a good pull request! Didn't find anything to keep it from further merging!

File: ClientApp/ClientApp/Coordinator/Coordinators/InitializationCoordinator.swift [hunk 1]
```diff
@@ -5,11 +5,11 @@ final class InitializationCoordinator: BaseCoordinator {
     var onAuthNavigate: EmptyCompletion?
 
     lazy var startController: UIViewController = {
-        let vm = OnboardingViewModelImpl()
-        vm.onAuthNavigate = onAuthNavigate
-        let vc = OnboardingViewController(viewModel: vm)
-        router.setRootModule(vc, hideBar: true)
-        return vc
+        let viewModel = OnboardingViewModelImpl()
```
Comment (siuzanna) [inline ClientApp/ClientApp/Coordinator/Coordinators/InitializationCoordinator.swift:8]: remove Iml

File: ClientApp/ClientApp/Coordinator/Coordinators/TabbarItemsCoordinators/ProfileCoordinator.swift [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import UIKit
+
+final class ProfileCoordinator: BaseCoordinator {
+    var onCourses: EmptyCompletion?
+
+    private var mainVC: MainViewController!
```
Comment (siuzanna) [inline ClientApp/ClientApp/Coordinator/Coordinators/TabbarItemsCoordinators/ProfileCoordinator.swift:6]: viewController

File: ClientApp/ClientApp/MainScreen/MainViewController.swift [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+//
+//  Created by Alisher on 13.02.2024.
+//
+
+import Foundation
+import UIKit
+import CustomUI
+
+class MainViewController: UIViewController {
+    lazy var mainView = MainView()
```
Comment (siuzanna) [inline ClientApp/ClientApp/MainScreen/MainViewController.swift:13]: private

File: ClientApp/ClientApp/MainScreen/MainViewController.swift [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+//  ClientApp
+//
+//  Created by Alisher on 13.02.2024.
+//
+
+import Foundation
+import UIKit
+import CustomUI
+
+class MainViewController: UIViewController {
```
Comment (siuzanna) [inline ClientApp/ClientApp/MainScreen/MainViewController.swift:12]: final

File: ClientApp/ClientApp/RegisterScreen/AuthenticationViewController.swift [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+//
+//  Created by Alisher on 02.02.2024.
+//
+
+import Foundation
+import UIKit
+import CustomUI
+
+class AuthenticationViewController: UIViewController {
+    lazy var registerView = AuthenticationView()
```
Comment (siuzanna) [inline ClientApp/ClientApp/RegisterScreen/AuthenticationViewController.swift:13]: private

File: ClientApp/ClientApp/RegisterScreen/AuthenticationViewController.swift [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    }
+
+    private func addTargets() {
+        registerView.loginSegmentedControl.addTarget(self, action: #selector(segmentedControlValueChanged), for: .valueChanged)
+        registerView.getCodeButton.addTarget(self, action: #selector(getCode(_:)), for: .touchUpInside)
+        registerView.sendCodeAgainButton.addTarget(self, action: #selector(resendCode(_:)), for: .touchUpInside)
+        registerView.backButton.addTarget(self, action: #selector(segmentedControlValueChanged), for: .touchUpInside)
+    }
+
+    private func addDelegates() {
```
Comment (siuzanna) [inline ClientApp/ClientApp/RegisterScreen/AuthenticationViewController.swift:43]: setup

File: ClientApp/ClientApp/RegisterScreen/AuthenticationViewController.swift [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+    init(viewModel: AuthenticationViewModelProtocol) {
+        registerViewModel = viewModel
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func addTargets() {
```
Comment (siuzanna) [inline ClientApp/ClientApp/RegisterScreen/AuthenticationViewController.swift:36]: setup

File: ClientApp/ClientApp/RegisterScreen/AuthenticationViewController.swift [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+        registerViewModel.viewDelegate = self
+    }
+
+    @objc func segmentedControlValueChanged() {
+        let selectedIndex = registerView.loginSegmentedControl.selectedSegmentIndex
+        registerViewModel.onWhichView = (selectedIndex == 0) ? .signIn : .signUp
+    }
+
+    @objc func getCode(_ button: UIButton) {
+        if button.titleLabel?.text == R.string.localizable.getCode() {
```
Comment (siuzanna) [inline ClientApp/ClientApp/RegisterScreen/AuthenticationViewController.swift:57]: enum

File: ClientApp/ClientApp/RegisterScreen/View/AuthenticationView.swift [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+        addSubview(contentView)
+        addSubview(getCodeButton)
+        addSubview(sendCodeAgainButton)
+    }
+
+    func changeView(to view: OnWhichView) {
+        contentView.arrangedSubviews.forEach { $0.isHidden = true }
+        clearView(view: view)
+        backButton.isHidden = !(view == .code)
+        headerLabelView.font = R.font.poppinsBold(size: view == .code ? 20 : 32)
```
Comment (siuzanna) [inline ClientApp/ClientApp/RegisterScreen/View/AuthenticationView.swift:107]: static func peachiFont(ofSize size: CGFloat) -> UIFont? {
        return UIFont(name: "Peachi", size: size)
    }

File: CustomUI/CustomUI/UI constants/CustomColor.swift [hunk 10]
```diff
@@ -0,0 +1,20 @@
+//
+//  CustomColor.swift
+//  CustomUI
+//
+//  Created by Alisher on 05.02.2024.
+//
+
+import UIKit
+
+extension UIColor {
```
Comment (siuzanna) [inline CustomUI/CustomUI/UI constants/CustomColor.swift:10]: remove  public struct CustomColor {

**Pull request merged**